### PR TITLE
Added FXIOS-6822 [v116] hook up password flow

### PR DESCRIPTION
**Pull request by OrlaM**
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-6822)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/15194)

### Description
Hook up the navigation logic in the password manager screens

### Pull requests checks where applicable
- [x] Fill in the three TODOs above (tickets number and description of your work)
- [x] The PR name follows our [naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Accessibility implemented and tested (minimum Dynamic Text and VoiceOver)
- [x] Unit tests written and passing
- [ ] Documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by OrlaM** [comment]
Updating to include [FXIOS-6486](https://mozilla-hub.atlassian.net/browse/FXIOS-6486)
#14574 
I accidentally removed the screens reliance on this when the flag is on, so I just updated to remove the reference.

File: Client/Coordinators/SettingsCoordinator.swift [hunk 1]
```diff
@@ -256,4 +261,11 @@ class SettingsCoordinator: BaseCoordinator, SettingsDelegate, SettingsFlowDelega
     func pressedTheme() {
         router.push(ThemeSettingsController())
     }
+
+    // MARK: - PasswordManagerCoordinatorDelegate
+
+    func didFinishPasswordManager(from coordinator: PasswordManagerCoordinator) {
```
Comment (lmarceau) [inline Client/Coordinators/SettingsCoordinator.swift:267]: Can we have a test for this?

File: Client/Coordinators/PasswordManagerCoordinator.swift [hunk 2]
```diff
@@ -22,9 +36,40 @@ class PasswordManagerCoordinator: BaseCoordinator, PasswordManagerFlowDelegate {
...
+    func showPasswordOnboarding() {
+        let viewController = PasswordManagerOnboardingViewController()
+        viewController.coordinator = self
+        router.push(viewController)
+    }
+
+    // MARK: - PasswordManagerFlowDelegate
 
-    func finishPasswordManagerFlow() {}
+    func continueFromOnboarding() {
```
Comment (lmarceau) [inline Client/Coordinators/PasswordManagerCoordinator.swift:54]: I think there's a couple of methods here that could get some tests as well

File: Client/Frontend/PasswordManagement/PasswordManagerListViewController.swift [hunk 3]
```diff
@@ -168,7 +170,9 @@ class PasswordManagerListViewController: SensitiveViewController, Themeable {
         super.viewDidAppear(animated)
 
         guard settingsDelegate != nil else {
-            if CoordinatorFlagManager.isCoordinatorEnabled {
+            if CoordinatorFlagManager.isSettingsCoordinatorEnabled {
+                return
```
Comment (lmarceau) [inline Client/Frontend/PasswordManagement/PasswordManagerListViewController.swift:174]: 🎉

**Pull request merged**